packages/astro/src/integrations/index.ts

```diff
@@ -138,6 +153,11 @@ export async function runHookConfigSetup({
 			) {
 				addContentEntryType(mdxContentEntryType);
 			}
+
+			// Add custom client directives to settings, waiting for compiled code by esbuild
+			for (const [name, compiled] of addedClientDirectives) {
+				updatedSettings.clientDirectives.set(name, await compiled);
```

Comment:
Any perf concerns that waiting on these slows down dev startup time? What would be the downsides to saving the Promise instead? Code complexity is one thing that I would imagine.

Reply:
Yeah the first few tries I went with a thunk and promise approach, and code complexity was the main issue that I continue that.
1. It async poisons the renderer.
2. It touches a lot of types (the map has to be passed/drilled down to the renderer).

That was also why I kept prebundling the builtin directives so default apps don't have to always build it. But for custom directives, since we recommend keeping it small, I think the bundling is fast enough that it won't be noticable too.

Also, since we prebundle the custom directives eagerly above, this `for await` pattern should execute similarly to a `Promise.all`